client/my-sites/site-settings/form-general.jsx

```diff
@@ -352,51 +348,6 @@ class SiteSettingsFormGeneral extends Component {
 		);
 	}
 
-	dateTimeFormat() {
```

Comment:
In addition to removing date & time format from General settings, let's also remove:

* The `date-time-format` route from `client/my-sites/site-settings/index.js`
* `DateTimeFormat` and `dateTimeFormat` from `client/my-sites/site-settings/controller.js`

because they'll no longer be necessary as we manage these settings inline. And we don't want to include the date & time stuff into multiple chunks, right? 😉